hlf-api-matriculation.md

```diff
@@ -0,0 +1,135 @@
+# Hyperledger Matriculation Api
+
+In the following, the hlf-api for the "matriculation data" - handling will be explained.
+For insight in the General hlf-api, please refer to it's .Readme - File
+
+## Methods
+
+Here the possible Actions are presented, that can be invoked on the hyperledger-network.
+Any Method can at any point return 2 types of errors. 
+  - TransactionError(methodName, JsonError) :: Something went wrong during the action. The Model recognized an invalidity.
+  - HyperledgerError(methodName, Exception) :: Something went wrong with the Framework. Hyperledger threw an internal error.
+
+### Add Matriculation Data
+- Pass (*transient*)
+    - MatriculationData :: Json (refer to: [MatriculationData](#matriculationData))
+- Receive
+    - "" :: String
+        => Success
+    - TransactionError :: Json (refer to: [DetailedError](#detailedError---example))
+        => error is returned
+          If the given parameters could not be parsed.
+            If the string could not be parsed as json, the list of invalidParams is empty.
```

Comment:
If by string you mean the json matriculation data, this would return a `GenericError` instead of a `DetailedError` with empty parameter list (at least on the chaincode side). Just noticed that this is also still wrong at [hyperledger_matriculation_api.md](https://github.com/upb-uc4/api/blob/develop/hyperledger_matriculation_api.md).
```suggestion
```

Reply:
Do not commit these suggestions, this is about to change...

Reply:
resolved with new commits